src/persist-client/src/bin/persist_open_loop_benchmark.rs

```diff
@@ -178,11 +175,10 @@ async fn run(args: Args) -> Result<(), anyhow::Error> {
         let start = start.clone();
         let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel();
 
-        let data_generator_handle = mz_ore::task::spawn_blocking(
-            || format!("data-generator-{}", idx),
-            move || {
+        let data_generator_handle =
+            mz_ore::task::spawn(|| format!("data-generator-{}", idx), async move {
```

Comment:
I think Ruchir was using a blocking task on purpose here, to ensure that the data generator got it's own CPU thread that it can use use as it wants. Not sure that reasoning is sound, though, and it's definitely nicer when we can use `async` code in the block.

Reply:
Hmm then maybe it should get its own Runtime or even just a Thread? The old impl was busy-waiting which also has its downsides

Reply:
I thought about this more and neither of those seem to be the "tokio way" (which I'm gradually learning to just accept and embrace). I think the problem we might hit here in practice is that the data generation call doesn't have any yield points. So, probably just that bit needs to be moved into a "blocking" call

Reply:
Meaning, move everything after this one `async` call of into a `spawn_blocking`? The whole generator code only had this one `await` on the barrier. Side note: I don't know how useful having the barriers for synchronization is here.